File: src/types/warehouse-item.ts
export interface WarehouseItem {
  id: string;
  creator_id: string;
  feed_item_id: string | null;
  title: string;
  file_path: string;
  thumbnail_path: string | null;
  platform: string | null;
  original_url: string | null;
  published_at: string | null;
  duration: number | null;
  file_size: number;
  imported_at: string;
  is_manual_import: boolean;
}

export interface CreateWarehouseItemRequest {
  creator_id: string;
  feed_item_id?: string | null;
  title: string;
  file_path: string;
  thumbnail_path?: string | null;
  platform?: string | null;
  original_url?: string | null;
  published_at?: string | null;
  duration?: number | null;
  file_size: number;
  is_manual_import: boolean;
}

File: src-tauri/src/models/warehouse_item.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct WarehouseItem {
    pub id: String,
    pub creator_id: String,
    pub feed_item_id: Option<String>,
    pub title: String,
    pub file_path: String,
    pub thumbnail_path: Option<String>,
    pub platform: Option<String>,
    pub original_url: Option<String>,
    pub published_at: Option<String>,
    pub duration: Option<i64>,
    pub file_size: i64,
    pub imported_at: String,
    pub is_manual_import: bool,
}

#[derive(Debug, Deserialize)]
pub struct CreateWarehouseItemRequest {
    pub creator_id: String,
    pub feed_item_id: Option<String>,
    pub title: String,
    pub file_path: String,
    pub thumbnail_path: Option<String>,
    pub platform: Option<String>,
    pub original_url: Option<String>,
    pub published_at: Option<String>,
    pub duration: Option<i64>,
    pub file_size: i64,
    pub is_manual_import: bool,
}
